internal/workflow.go

```diff
@@ -692,25 +690,24 @@ func (wc *workflowEnvironmentInterceptor) ExecuteChildWorkflow(ctx Context, chil
 		}
 	}, func(r WorkflowExecution, e error) {
 		if e == nil {
-			childWorkflowExecution = &r
+			if cancellable {
+				cancellationCallback.fn = func(v interface{}, _ bool) bool {
+					if ctx.Err() == ErrCanceled && !mainFuture.IsReady() {
+						// child workflow started, and ctx cancelled
+						getWorkflowEnvironment(ctx).RequestCancelChildWorkflow(options.Namespace, r.ID)
+					}
+					return false
+				}
+				_, ok, more := ctxDone.receiveAsyncImpl(cancellationCallback)
+				if ok || !more {
+					cancellationCallback.fn(nil, more)
+				}
+			}
 		}
+
 		executionSettable.Set(r, e)
 	})
 
-	if cancellable {
-		cancellationCallback.fn = func(v interface{}, more bool) bool {
-			if ctx.Err() == ErrCanceled && childWorkflowExecution != nil && !mainFuture.IsReady() {
```

Comment:
Will this work if we just remove the `childWorkflowExecution != nil check`?

Reply:
added a comment explaining why we are waiting on initiation